packages/connect-core/src/entities/App.ts

```diff
@@ -26,81 +26,91 @@ export interface AppData {
 }
 
 export default class App extends CoreEntity implements AppData {
-  readonly abi?: Abi
-  readonly address!: string
-  readonly appId!: string
-  readonly appName?: string
-  readonly author?: string
-  readonly codeAddress!: string
-  readonly contentUri?: string
-  readonly contractPath?: string
-  readonly deprecatedIntents?: { [version: string]: AppIntent[] }
-  readonly description?: string
-  readonly htmlUrl?: string
-  readonly intents?: AppIntent[]
-  readonly icons?: { src: string; sizes: string }[]
-  readonly isForwarder?: boolean
-  readonly isUpgradeable?: boolean
-  readonly kernelAddress!: string
-  readonly name?: string
-  readonly registryAddress!: string
-  readonly registry?: string
-  readonly repoAddress?: string
-  readonly sourceUrl?: string
-  readonly version?: string
+  #created!: boolean
+  abi?: Abi
+  address!: string
+  appName?: string
+  appId!: string
+  author?: string
+  codeAddress!: string
+  contentUri?: string
+  contractPath?: string
+  deprecatedIntents?: { [version: string]: AppIntent[] }
+  description?: string
+  icons?: { src: string; sizes: string }[]
+  intents?: AppIntent[]
+  isForwarder?: boolean
+  isUpgradeable?: boolean
+  htmlUrl?: string
+  kernelAddress!: string
+  name?: string
+  registry?: string
+  registryAddress!: string
+  repoAddress?: string
+  sourceUrl?: string
+  version?: string
 
-  constructor(
-    { artifact, manifest, ...data }: AppData,
-    connector: ConnectorInterface
-  ) {
+  constructor(connector: ConnectorInterface) {
     super(connector)
 
-    // TODO: If no metadata, fallback to resolve ourselves with ipfs
+    this.#created = false
+  }
+
+  get created(): boolean {
+    return this.#created
+  }
+
+  async create({ artifact, manifest, ...data }: AppData): Promise<void> {
```

Comment:
For `create`, I was more thinking of a static method, that would fetch everything needed and instantiate, so we can do this:

```jsx
const app = await App.create(/* … */)
```

Rather than this:

```jsx
const app = new App(/* … */)
await app.create()
```

That way, `App` would never be in a state where it is instantiated, but not initialized.

Reply:
Oh, I didn't think about that. I'll include this change as a new PR.